### DISCUSSION TITLE
How to render a directory of quarto documents to be included within a website?

### DISCUSSION THREAD

**Discussion by Wcarcamo** [quarto-dev/quarto-cli#8250] [Q&A] [answered]
### Description

I would like to include a directory of quarto documents to be rendered automatically when I run `$ quarto render` for a website project. 

Its a simple website where my index is a dashboard that pulls data from a csv file and displays it in an interactive reactable table. 

The table generates custom `<a>` tags with relative links pointing to the rendered quarto html files within the directory in question.

Below is reproducible example.

Quarto Version: 1.4.530

Website project layout:

```
📦 Quarto Project
|-- 📂 subdirectory
|   |-- 📄 doc1.qmd
|   `-- 📄 doc2.qmd
|-- 📄 _quarto.yml
|-- 📄 index.qmd
`-- 📄 data.csv
```

`index.qmd`

````qmd
---
format: dashboard
---

```{r setup}
#| echo: false
library(tidyverse)
library(reactable)

dat <- rio::import("data.csv")
```

```{r data}
dat |>
  reactable(
    searchable = T, 
    compact = TRUE,
    columns = list(
      column5 = colDef(show = FALSE),
      column4 = colDef(
        cell = function(value, index) {
          htmltools::tags$a(
            href = dat[index, "column5"],
            target = "_blank",
            as.character(value)
          )
        }
      )
    )
  )
```
````

`data.csv`

```csv
column1, column2, column3, column4, column5
valueA1, valueA2, valueA3, valueA4, "./subdirectory/doc1.html"
valueB1, valueB2, valueB3, valueB4, "./subdirectory/doc2.html"
```

The problem is when I try to render the website project none of the quarto documents under the `subdirectory` get rendered or included within my rendered site. 

I've discovered that you can use `render` option under the website quarto [Project reference](https://quarto.org/docs/reference/projects/websites.html#project) to include specific quarto documents for a website. 

For example if I update my `_qaurto.yml` file as so:

```
project:
  type: website
  render: 
    - index.qmd
    - subdirectory/doc1.qmd
```

Then my link to `./subdirectory/doc1.html` works as expected, but `./subdirectory/doc2.html` does not until I add it to the render list in the yaml.

I expect to have about 200 unique pages under `subdirectory` and I don't want to individually add them all in the yaml.
Is there a way to include a whole directory.

I've tried the following in the yaml but with no success:

```
render: 
  - subdirectory/
  - subdirectory/*
  - subdirectory/*.qmd
```

Appreciate the help 🙏

**Comment by mcanouil** [comment]
Are you making a website (pages are interlinked) or do you want to have a multiple documents project?

Can you share an example as a Git repository?

**Reply by Wcarcamo** [reply]
I am making a website. Not sure what you are asking about for multiple documents project.

I've created an example repo found [here](https://github.com/Wcarcamo/quarto-example).

Sorry for the delay in response, was off for 3 day weekend.

**Reply by mcanouil** [reply, marked as answer]
Few things:
- if you want to render everything, no need to set `render` because it means you have to add all the targets.
- ` ```{r setup} ` is inline R approach to set label, while you are using YAML style for `echo: false`. Do not mix syntaxes, *i.e.*, use YAML style for everything: `label: setup`.

Everything works as expected based on your setup from https://github.com/Wcarcamo/quarto-example

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/7ea206b3-f5da-4f79-8686-412eb327a32d">

Using wildcard:

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/510a5e9d-212a-4efb-8c92-ad8b24990c78">

Without defining the render targets:

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/9d3d4c8e-d2a7-4117-b30a-cfbd6f1bc734">


Note that I deleted `.quarto` and. `_site` between each render.

**Reply by Wcarcamo** [reply]
I swear it wasn't rendering before, I tried my work and personal computer...but after cloning the example I'm getting the same results you are.

I went back to test everything that was different in my actual website and I found where the issue comes from. I had a leading underscore in the name of my subdirectory, i.e. `_subdirectory`. Not sure why but when I render the project with this naming convention, nothing in the subdirectory would be included and my links would break. 

Also my bad about labels, old habits die hard. Thank you for your help!

**Reply by mcanouil** [reply]
Files and directories starting with underscore are hidden for Quarto.

This is stated in a note in <https://quarto.org/docs/websites/index.html#render-targets>.

**Reply by Wcarcamo** [reply]
Wow I completely missed that, now I'm embarrassed haha thanks again for the info
